geodetic_utils/src/imu_compass_asl.cpp

```diff
@@ -302,7 +302,17 @@ void IMUCompass::repackageImuPublish(tf::StampedTransform transform)
   sensor_msgs::Imu publish_imu;
   o_imu_reading = tf::Transform(new_quaternion, tf::Vector3(0, 0, 0));
   o_imu_reading = o_imu_reading * (transform.inverse());
-  tf::quaternionTFToMsg(o_imu_reading.getRotation(), curr_imu_reading_->orientation);
+
+  // TODO: hack, rotate resulting heading measurement
+  double hroll, hpitch, hyaw;
+  tf::Matrix3x3(o_imu_reading.getRotation()).getRPY(hroll, hpitch, hyaw);
+
+  //std::cout << "rpy: " << hroll << ", " << hpitch << " ," << hyaw << std::endl;
+
+  tf::Quaternion hackq = tf::createQuaternionFromRPY(hroll, hpitch, -(hyaw+M_PI/2));
```

Comment:
Changed, this is because we want 0 deg to be associated with East, rather than North. Easily explained in person on Monday.